packages/deployments/contracts/contracts/core/messaging/interfaces/IConnector.sol

```diff
@@ -40,11 +40,23 @@ import {IProposedOwnable} from "../../shared/interfaces/IProposedOwnable.sol";
  * - bsc (use multichain for messaging)
  */
 interface IConnector is IProposedOwnable {
-  function sendMessage(bytes memory _data) external;
+  /**
+   * @notice This should be called by RootManager to propagate an updated aggregate root on the Hub/L1
+   * chain out to the spoke/L2 chains.
+   */
+  function propagateRoot(bytes memory _data) external;
 
-  function processMessage(address _sender, bytes memory _data) external;
+  /**
+   * @notice This is called by relayers to trigger passing of current root to mainnet root manager
+   * @dev This is called at specific time intervals
+   */
+  function sendRoot() external;
```

Comment:
I don't love having two different functions that are designed to both send a `bytes32` to their mirror. it seems like we could make it a single function and use the `IS_HUB` tag

Reply:
is the issue the difference in args? if so, then lets be more explicit in the name about the roots we are sending. `sendOutboundRoot` / `sendAggregateRoot` or something

Reply:
down for that! the second thing, that is. `sendOutboundRoot` and `sendAggregateRoot`